tubular/base.py

```diff
@@ -167,7 +172,7 @@ def _combine_X_y(self, X: pd.DataFrame, y: pd.Series) -> pd.DataFrame:
                 stacklevel=2,
             )
 
-        X_y = X.copy()
+        X_y = X[X.columns]
```

Comment:
S: Would `pd.assign` work here? A quick plan in codespaces suggests that ti's much more efficient:

```python
import timeit
import pandas as pd

pd.options.mode.copy_on_write = True

df = pd.DataFrame({"student_id": [1, 2, 3], "grade": ["A", "C", "D"]})
y = pd.Series(['a', 'b', 'c'])

time_assign_col = timeit.timeit("X_y = df[df.columns]; X_y['_temporary_response'] = y.to_numpy()", globals=globals(), number=1000)
print(f'{time_assign_col*1000:.3f} microseconds/run')
# 355.369 microseconds/run

time_assign_col = timeit.timeit("X_y2 = df.assign(_temporary_response = y)", globals=globals(), number=1000)
print(f'{time_assign_col*1000:.3f} microseconds/run')
# 173.598 microseconds/run
# Twice as fast - I suspect that assign doesn't copy

X_y = df[df.columns]; X_y['_temporary_response'] = y.to_numpy()
print(X_y)
#   student_id grade _temporary_response
#0           1     A                   a
#1           2     C                   b
#2           3     D                   c

X_y2 = df.assign(_temporary_response = y)
print(X_y2)
#   student_id grade _temporary_response
#0           1     A                   a
#1           2     C                   b
#2           3     D                   c

print(df)
#   student_id grade
#0           1     A
#1           2     C
#2           3     D
```

No updates by reference - references are behaving themselves.

Reply:
Updated to use df.assign for temporary response field [here](https://github.com/lvgig/tubular/pull/197/commits/e15f94976aeb13ef5547b537de31b14e5f27bf28)